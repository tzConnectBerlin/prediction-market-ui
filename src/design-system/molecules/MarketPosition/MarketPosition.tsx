import React from 'react';
import { Grid } from '@material-ui/core';
import styled from '@emotion/styled';
import { lightTheme as theme } from '../../../theme';
import { Typography } from '../../atoms/Typography';

interface StyledLabelProps {
  icon?: string;
  fontColor?: string;
}

const StyledGrid = styled(Grid)`
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
`;

const StyledLabel = styled.div<StyledLabelProps>`
  color: ${({ fontColor }) => fontColor};
  padding: 0.2em;
  &.hasIcon {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    & > * {
      margin-left: 0.2em;
      font-size: 1.5em;
    }
  }
`;
export interface MarketPositionProps {
  tokenList?: { type: string; value?: number }[];
}

export const MarketPosition: React.FC<MarketPositionProps> = ({ tokenList }) => {
  if (!tokenList) {
    return null;
  }
  const getTokenList = () => {
    return tokenList?.map((item, i) => {
      return (
        <Grid container item xs={6} key={i} direction="row" justifyContent="space-between">
          <StyledLabel fontColor={theme.palette.text.secondary}>{item.type}</StyledLabel>
          <StyledLabel>{item.value}</StyledLabel>
        </Grid>
      );
    });
  };

  return (
    <StyledGrid container spacing={1}>
      <Typography size="h6" marginLeft="1rem">
        Position Summary
      </Typography>
      {/* add translation */}
      <Grid container item xs={12} spacing={1} direction="column" marginRight="1rem" marginLeft="0">
        {getTokenList()}
      </Grid>
    </StyledGrid>
  );
};
