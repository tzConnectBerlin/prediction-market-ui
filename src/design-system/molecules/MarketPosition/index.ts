export { MarketPosition as default, MarketPosition } from './MarketPosition';
export type { MarketPositionProps } from './MarketPosition';
