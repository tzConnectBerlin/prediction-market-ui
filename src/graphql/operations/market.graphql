subscription Market {
  markets: allPmmStorageMarketMapOrderedsList {
    id
    marketId: idxMarketsNat
    metadataIpfsHash
    metadataDescription
    metadataAdjudicator
    marketsState
    txContext: publicTxContextByTxContextId {
      blockInfo: publicLevelByLevel {
        block: level
        bakedAt
      }
      operationGroupNumber
      operationNumber
      contentNumber
    }
    marketInfo: pmmStorageMarketMapById {
      marketBootstrapped: pmmStorageMarketMapMarketbootstrappedsByStorageMarketMapIdList {
        auctionRewardCurrencyPool: currencyPoolCreatorRewardCurrencyPool
        liquidityRewardPool: currencyPoolCreatorRewardCurrencyPool
        marketCurrencyPool: currencyPoolMarketCurrencyPool
        bootstrapYesProbability: marketBootstrappedBootstrapYesProbability
        winningPrediction: resolutionWinningPrediction
        resolutionResolvedAtBlock
        marketBootstrappedBootstrappedAtBlock
      }
      auctionRunning: pmmStorageMarketMapAuctionrunningsByStorageMarketMapIdList {
        auctionRunningAuctionPeriodEnd
        auctionRunningQuantity
        auctionRunningYesPreference
        auctionRunningUniswapContribution
      }
    }
  }
}
