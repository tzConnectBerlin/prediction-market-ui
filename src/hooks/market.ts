import { useParams } from 'react-router-dom';
import { MARKET_ADDRESS } from '../globals';

interface MarketPathParams {
  marketAddress?: string;
  questionHash?: string;
}

export const useMarketPathParams = (): MarketPathParams => {
  const { marketAddress, questionHash } = useParams<MarketPathParams>();
  const mrktAddr = marketAddress === 'main' ? MARKET_ADDRESS : marketAddress;
  return {
    marketAddress: mrktAddr,
    questionHash,
  };
};
