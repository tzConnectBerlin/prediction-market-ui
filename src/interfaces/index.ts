export * from './wallet';
export * from './router';
