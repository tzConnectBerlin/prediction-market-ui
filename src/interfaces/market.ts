import { MichelsonMap } from '@taquito/taquito';
import { BigNumber } from 'bignumber.js';

export type QuestionType = string;

export interface Links {
  label: string;
  url: string;
}

export enum MarketStateType {
  auctionRunning = 'auction',
  marketBootstrapped = 'market',
}

export enum Role {
  participant = 'Participant',
  adjudicator = 'Adjudicator',
}

export interface AuctionNode {
  auctionRunningAuctionPeriodEnd: string;
  auctionRunningQuantity: string;
  auctionRunningYesPreference: string;
  auctionRunningUniswapContribution: string;
}

export interface MarketNode {
  auctionRewardCurrencyPool: string;
  liquidityRewardPool: string;
  marketCurrencyPool: string;
  bootstrapYesProbability: string;
  liquidityRewardSupplyUpdatedAtBlock: string;
  winningPrediction?: string;
  resolutionResolvedAtBlock?: number;
  marketBootstrappedBootstrappedAtBlock: string;
}

export interface PortfolioMarket {
  question: (string | undefined)[];
  holdings: string[] | string;
  price: string[] | string;
  total: string[] | string;
}

export interface PortfolioAuction {
  question: string;
  probability: string;
  quantity: string;
}

export interface MarketStateNode {
  node: AuctionNode | MarketNode;
}

export interface StorageMarketMapMarketBootstrappeds {
  nodes: MarketStateNode[];
}

export interface StorageMarketMapAuctionRunnings {
  nodes: MarketStateNode[];
}

export interface GraphMarketNode {
  node: GraphMarket;
}
export interface GraphMarket {
  id: number;
  block: number;
  deleted: boolean;
  marketId: string;
  metadataIpfsHash: string;
  metadataDescription: string;
  metadataAdjudicator: string;
  state: string;
  dateTime: {
    bakedAt: string;
  };
  storageMarketMapAuctionRunnings: StorageMarketMapAuctionRunnings;
  storageMarketMapMarketBootstrappeds: StorageMarketMapMarketBootstrappeds;
}

export interface IPFSMarketData {
  question: QuestionType;
  auctionEndDate: string;
  iconURL?: string;
  ticker: string;
}

export interface Market extends Partial<AuctionNode>, Partial<MarketNode>, IPFSMarketData {
  marketId: string;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  state: MarketStateType;
  yesPrice: number;
  liquidity?: number | string;
  block: number;
  bakedAt: string;
}

export interface AllMarkets {
  markets: {
    marketNodes: GraphMarket[];
  };
}

export interface AllLedgers {
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface AllMarketsLedgers {
  markets: {
    marketNodes: GraphMarket[];
  };
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface MarketCardStatistic {
  type: string;
  value: string | number;
  changes?: string;
  currency?: Currency;
  tokenType?: TokenType;
}

export interface MarketCardToken {
  type: TokenType;
  value: number | string;
}

export interface MarketCardData extends Market {
  tokens?: MarketCardToken[];
  statistics?: MarketCardStatistic[];
}

export interface CreateMarket {
  marketId: number;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  auctionEnd: string;
  initialBid: number;
  initialContribution: number;
  tokenType: 'fa12' | 'fa2';
  tokenAddress: string;
}

export interface GraphBet {
  probability: string;
  quantity: string;
}

export interface Bet {
  block: number;
  originator: string;
  probability: number;
  quantity: number;
  marketId: string;
}

export interface BetEdge {
  bet: GraphBet;
}

export interface GraphBets {
  totalBets: number;
  betEdges: BetEdge[];
}

export interface LqtProviderNode {
  id: number;
  marketId: string;
  originator: string;
  bets: GraphBets;
  block: number;
}

export interface LqtProviderEdge {
  lqtProviderNode: LqtProviderNode;
}

export interface StorageLiquidityProviderMaps {
  lqtProviderEdge: LqtProviderEdge[];
}

export interface AllBets {
  storageLiquidityProviderMaps: StorageLiquidityProviderMaps;
}

export enum MarketTradeType {
  payIn = 'payIn',
  payOut = 'payOut',
}

export enum FormType {
  buy = 'Buy',
  sell = 'Sell',
  addLiquidity = 'Add Liquidity',
  removeLiquidity = 'Remove Liquidity',
}

export interface LiquidityValues {
  probability: number;
  quantity: number;
}

export interface TokenSupplyMap {
  id: number;
  tokenId: string;
  totalSupply: string;
  tokenReserve: string;
  deleted: boolean;
}

export interface StorageSupplyMaps {
  supplyMaps: TokenSupplyMap[];
}

export interface AllTokens {
  storageSupplyMaps: StorageSupplyMaps;
}

export enum TokenType {
  yes = 'Yes',
  no = 'No',
}

export interface LedgerMap {
  id: number;
  block: number;
  deleted: boolean;
  owner: string;
  tokenId: string;
  quantity: string;
}

export interface StorageLedgerMaps {
  ledgerMaps: LedgerMap[];
}

export interface Token {
  id: number;
  tokenId: string;
  quantity: string;
  block: number;
  dateTime: {
    bakedAt: string;
  };
}

export interface TokenQuantity {
  token: Token[];
}

export interface MarketPricePoint {
  yesPrice: number;
  block: number;
  bakedAt: string;
}

export interface AddressTokens {
  tokenQuantity: TokenQuantity;
}

export interface AuctionMarkets {
  [key: string]: Market[];
}

// TODO: clean the stuff below

export interface QuestionState {
  [key: number]: {
    [key in MarketStateType]: symbol;
  };
}

export enum Currency {
  USD = '$',
  EUR = 'â‚¬',
}

export type CurrencyTypes = keyof typeof Currency;

export enum MarketEntrypoint {
  Bid = 'bid',
  BuyYes = 'buyYes',
  BuyNo = 'buyNo',
  ClaimWinnings = 'claimWinnings',
  CloseAuction = 'closeAuction',
  CreateQuestion = 'createQuestion',
  WithdrawAuction = 'withdrawAuction',
}

export interface CreateQuestion {
  question: QuestionType;
  auctionEndDate: Date | string;
  marketCloseDate: Date | string;
  iconURL?: string;
  yesAnswer: string;
  rate?: number;
  quantity?: number;
}

export interface Bid {
  question: QuestionType;
  rate: number;
  quantity: number;
}

export interface BuyToken {
  tokenType: TokenType;
  question: QuestionType;
  quantity: number;
}

export interface CloseMarket {
  tokenType: TokenType;
  question: QuestionType;
}

export interface ClaimWinnings {
  question: QuestionType;
}

export interface QuestionMetaData extends CreateQuestion {
  hash: string;
}

export interface BidEntry {
  rate: number;
  quantity: number;
  total_token: number;
}

export type BidRegistry = MichelsonMap<string, BidEntry>;

export interface QuestionEntry {
  owner: string;
  state: QuestionState;
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistry;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: BigNumber;
    no_token_id: BigNumber;
    lqt_token_id: BigNumber;
  };
  answer?: string | null;
  winning_token?: BigNumber | null; // token that will receive the reward
  total_auction_quantity: BigNumber;
  yes_preference: BigNumber;
  uniswap_contribution_factor: BigNumber;
}

export interface AuctionData {
  yes: number;
  no: number;
  participants: number;
}

export interface AuctionDataMap {
  [key: string]: AuctionData;
}

export interface MarketFilter {
  auctions: boolean;
  allMarkets: boolean;
  openMarkets: boolean;
  closedMarkets: boolean;
  onlyMyMarkets: boolean;
}

/**
 * Temp middleware interfaces
 */

export interface BidEntryMDW {
  rate: string;
  quantity: string;
  total_token: string;
}

export interface BidRegistryMDW {
  [key: string]: BidEntryMDW;
}

export interface QuestionEntryMDW {
  owner: string;
  state: {
    [key in MarketStateType]?: null;
  };
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistryMDW;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: string;
    no_token_id: string;
    lqt_token_id: string;
  };
  answer?: string | null;
  winning_token?: string | null;
  total_auction_quantity: string;
  yes_preference: string;
  uniswap_contribution_factor: string;
  price_yes?: string;
}

export interface QuestionEntryMDWMap {
  [key: string]: QuestionEntryMDW;
}

export interface StableCoinResponse {
  [key: string]: string;
}

export interface LedgerBalanceResponse {
  [tokenId: string]: { [address: string]: string };
}

export interface ContractError {
  [key: number]: string;
}

export interface DropDownItems {
  label: string;
  value: string | number;
}
