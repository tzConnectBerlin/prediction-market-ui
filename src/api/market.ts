import { getQuestionData } from '../contracts/Market';
import { AuctionData, AuctionDataMap, CreateQuestion, QuestionMetaData } from '../interfaces';
import { fetchIPFSData } from '../ipfs/ipfs';
import { divideDown } from '../utils/math';

export const getIPFSDataByKeys = async (questionKeys: string[]): Promise<QuestionMetaData[]> => {
  const data: QuestionMetaData[] = await Promise.all(
    questionKeys.map(async (hash) => {
      const questionData = await fetchIPFSData<CreateQuestion>(hash);
      return { hash, ...questionData };
    }),
  );
  return data;
};

export const getAuctionData = async (questionHashes: string[]): Promise<AuctionDataMap> => {
  const questionDetails = await getQuestionData(questionHashes);
  return Object.keys(questionDetails).reduce((acc: AuctionDataMap, hash: string) => {
    const item = questionDetails[hash];
    const yesVal = item.yes_preference.dividedToIntegerBy(item.total_auction_quantity).toNumber();
    let yes = divideDown(yesVal);
    if (Number.isNaN(yes)) {
      yes = 0.5;
    }
    const result: AuctionData = {
      no: Math.round((1 - yes + Number.EPSILON) * 100) / 100,
      yes: Math.round((yes + Number.EPSILON) * 100) / 100,
      participants: item.auction_bids.size,
    };
    return { ...acc, [hash]: result };
  }, {});
};
