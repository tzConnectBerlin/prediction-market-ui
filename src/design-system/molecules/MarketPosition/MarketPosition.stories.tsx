import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { MarketPosition, MarketPositionProps } from './MarketPosition';
import { TokenType } from '../../../interfaces/market';

export default {
  title: 'Molecule/MarketPosition',
  component: MarketPosition,
} as Meta;

const Template: Story<MarketPositionProps> = (args) => <MarketPosition {...args} />;

export const TokenAndStatisticList = Template.bind({});
TokenAndStatisticList.args = {
  tokenList: [
    {
      type: TokenType.yes,
      value: 14,
    },
    {
      type: TokenType.no,
      value: 5,
    },
  ],
};
