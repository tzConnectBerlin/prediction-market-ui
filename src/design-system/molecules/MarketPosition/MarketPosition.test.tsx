import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { TokenType } from '../../../interfaces/market';
import { MarketPosition, MarketPositionProps } from './MarketPosition';

const defaultProps: MarketPositionProps = {
  tokenList: [
    {
      type: TokenType.yes,
      value: 14,
    },
    {
      type: TokenType.no,
      value: 5,
    },
  ],
};

describe('Snapshot testing MarketPosition Component', () => {
  it('renders correctly without Props', () => {
    const cardContent = renderer.create(<MarketPosition />).toJSON();
    expect(cardContent).toMatchSnapshot();
  });

  it('renders correctly only tokenList', () => {
    const cardContent = renderer
      .create(<MarketPosition tokenList={defaultProps.tokenList} />)
      .toJSON();
    expect(cardContent).toMatchSnapshot();
  });
});
