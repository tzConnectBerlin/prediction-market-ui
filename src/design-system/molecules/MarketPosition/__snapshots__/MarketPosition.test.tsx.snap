// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketPosition Component renders correctly only tokenList 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-0>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  margin-left: 1rem;
}

.emotion-1.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-flex-basis: calc(100% + 8px);
  -ms-flex-preferred-size: calc(100% + 8px);
  flex-basis: calc(100% + 8px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 8px);
  margin-right: 1rem;
  margin-left: 0;
}

.emotion-2>.MuiGrid-item {
  max-width: none;
}

.emotion-2>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-2>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 {
  color: rgba(29, 34, 39, 0.54);
  padding: 0.2em;
}

.emotion-4.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-5.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-6 {
  padding: 0.2em;
}

.emotion-6.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-7.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-0"
>
  <h2
    className="MuiTypography-root MuiTypography-h2  emotion-1"
    lines={3}
  >
    Position Summary
  </h2>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-grid-xs-12 emotion-2"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-grid-xs-6 emotion-3"
    >
      <div
        className="emotion-4"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-5"
          lines={3}
        >
          Yes
        </h4>
      </div>
      <div
        className="emotion-6"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-7"
          lines={3}
        >
          14
        </h3>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-grid-xs-6 emotion-3"
    >
      <div
        className="emotion-4"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-5"
          lines={3}
        >
          No
        </h4>
      </div>
      <div
        className="emotion-6"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-7"
          lines={3}
        >
          5
        </h3>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing MarketPosition Component renders correctly without Props 1`] = `null`;
